### PR TITLE
[DROOLS-3188] Avoid cell merging on header

### PR DESCRIPTION
**Pull request by gitgabrio**
@kkufova @Rikkola @danielezonca 

Scope of this PR is to avoid horizontal cell merging on headers (https://issues.jboss.org/browse/DROOLS-3188).
More specifically, merging should be possible on
1) top group (i.e. only one "GIVEN" and one "EXPECT" cell)
2) at instance level (ie. columns of the same istance should have one single cell, for ex for "Book" or "Person") *but only inside the same group* (i.e. GIVEN/Person must not be merged with EXPECT/Person)
3) merging should *never* be possible at property level.

Please test also for regression on actual RunScenario

### CODE REVIEW

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioSimulationGridPanelClickHandler.java [hunk 1]
```diff
@@ -222,6 +222,9 @@ private boolean manageHeaderRightClick(ScenarioGrid scenarioGrid, int left, int
             return false;
         }
         String group = columnMetadata.getColumnGroup();
+        if (group.contains("-")) {
```
Comment (danielezonca) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioSimulationGridPanelClickHandler.java]: Can you please add a test?

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioSimulationGridPanelClickHandler.java [hunk 2]
```diff
@@ -342,6 +345,9 @@ protected boolean manageHeaderLeftClick(ScenarioGrid scenarioGrid, Integer uiCol
             return false;
         }
         String group = clickedScenarioHeaderMetadata.getColumnGroup();
+        if (group.contains("-")) {
```
Comment (danielezonca) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioSimulationGridPanelClickHandler.java]: Can you please add a test?

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/models/ScenarioGridModel.java [hunk 3]
```diff
@@ -376,10 +376,17 @@ public void updateHeader(int columnIndex, int headerRowIndex, String value) {
         }
         SimulationDescriptor simulationDescriptor = simulation.getSimulationDescriptor();
         FactMapping factMappingToEdit = simulationDescriptor.getFactMappingByIndex(columnIndex);
+        final FactIdentifier factIdentifierToEdit = factMappingToEdit.getFactIdentifier();
         if (editedMetadata.isInstanceHeader()) { // we have to update title and value for every column of the group
             IntStream.range(0, getColumnCount()).forEach(index -> {
                 FactMapping factMappingToCheck = simulationDescriptor.getFactMappingByIndex(index);
-                if (Objects.equals(factMappingToCheck.getFactIdentifier(), factMappingToEdit.getFactIdentifier())) {
+                final FactIdentifier factIdentifierToCheck = factMappingToCheck.getFactIdentifier();
```
Comment (danielezonca) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/models/ScenarioGridModel.java]: Can you please add a test?

**Comment by gitgabrio** [comment]
@manstis 
Hi! Could you please merge it and squash commit? Thanks!!!

**Pull request merged**